src/zope/tal/talinterpreter.py

```diff
@@ -92,27 +88,44 @@ def replaceAttrs(self, attrlist, repldict):
 
 
 class MacroStackItem(list):
-    # This is a `list` subclass for backward compability.
+    # This is a `list` subclass for backward compatibility.
     """Stack entry for the TALInterpreter.macroStack.
 
     This offers convenience attributes for more readable access.
 
     """
     __slots__ = ()
 
-    # These would be nicer using @syntax, but that would require
-    # Python 2.4.x; this will do for now.
+    @property
+    def macroName(self):
+        return self[0]
+
+    @property
+    def slots(self):
+        return self[1]
+
+    @property
+    def definingName(self):
+        return self[2]
+
+    @property
+    def extending(self):
+        return self[3]
+
+    @property
+    def i18nContext(self):
+        return self[5]
+
+    @property
+    def entering(self):
+        return self[4]
 
-    macroName = property(lambda self: self[0])
-    slots = property(lambda self: self[1])
-    definingName = property(lambda self: self[2])
-    extending = property(lambda self: self[3])
-    entering = property(lambda self: self[4],
-                        lambda self, value: operator.setitem(self, 4, value))
-    i18nContext = property(lambda self: self[5])
+    @entering.setter
+    def entering(self, value):
+        self[4] = value
```

Comment:
Move both the `entering` properties before `i18nContext` so that the properties stay in the same order as the item numbering?

Reply:
Implemented in https://github.com/zopefoundation/zope.tal/pull/19/commits/48e0bb2c669fe87795975d2fa6a04385521e4748.